src/Symfony/Component/Translation/Translator.php

```diff
@@ -419,8 +419,18 @@ private function getFallbackContent(MessageCatalogue $catalogue)
 
     private function getCatalogueCachePath($locale)
     {
+        $resources = array();
+        if (isset($this->resources[$locale])) {
+            foreach ($this->resources[$locale] as $resource) {
+                if (is_object($resource[1]) && !$resource[1] instanceof \Serializable) {
+                    $resource[1] = get_class($resource[1]);
```

Comment:
Using `spl_object_hash()` means that we get different paths on each request, doesn't it?. Isn't that an issue?